### PR TITLE
Fix Indexer use of $ne

### PR DESCRIPTION
**Pull request by haiodo**
Fix Indexer use of $ne

<sub><a href="https://front.hc.engineering/guest/platform?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NmIxM2I0OWU1NTRjZjllNTY4ZmEzZDMiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InBsYXRmb3JtIiwicHJvZHVjdElkIjoiIn0.5xX5DqUTtI7PiNicXch0qOR6bfsOZCLeRkclBpbjhCw">Huly&reg;: <b>UBERF-7815</b></a></sub>

### CODE REVIEW

File: server/core/src/fulltext.ts [hunk 1]
```diff
@@ -143,7 +145,7 @@ export class FullTextIndex implements WithFind {
 
     const ids: Set<Ref<Doc>> = new Set<Ref<Doc>>()
     const baseClass = this.hierarchy.getBaseClass(_class)
-    let classes = this.hierarchy.getDescendants(baseClass)
+    let classes = this.hierarchy.getDescendants(baseClass).filter((it) => !this.hierarchy.isMixin(it))
```
Comment (ThetaDR) [inline server/core/src/fulltext.ts:148]: Why filter for mixins is needed here?

Comment (haiodo) [reply server/core/src/fulltext.ts:148]: We could not have _class in DocIndexState to be one of mixins so they should not go into elastic, so we should skip them on search.

Comment (haiodo) [reply server/core/src/fulltext.ts:148]: Probable we need to add getBaseClass there, is a good point

Comment (ThetaDR) [reply server/core/src/fulltext.ts:148]: My concern is that it may filter out recruit.mixin.Candidate and skip it in the search. I might be wrong, though.

Comment (haiodo) [reply server/core/src/fulltext.ts:148]: Nope, it work fine with candidate, because of this.hierarchy.getBaseClass(_class)

**Review by ThetaDR — APPROVED** [review]

**Review by BykhovDenis — APPROVED** [review]

**Pull request merged**